File: src/main/java/net/mreunionlabs/tapgradle/generate/generator/JavaStructureGenerator.java
package net.mreunionlabs.tapgradle.generate.generator;

import net.mreunionlabs.tapgradle.generate.GeneratePluginExtension;

import java.io.File;

/**
 * Created by abangkis on 6/15/2016.
 */
public class JavaStructureGenerator {

    //TODO validate that java dir parent has been created
    public void createDirectories(GeneratePluginExtension ext) {
        File f = new File(ext.getJavaDir());
        f.mkdir();

        String packageString = ext.getPackageString();
        String[] split = packageString.split("\\.");

        File packageRoot = f;
        for (String dir : split) {
            if (dir != null && !"".equals(dir)) {
                packageRoot = new File(packageRoot, dir);
                packageRoot.mkdir();
            }
        }

        File pages = new File(packageRoot, "pages");
        pages.mkdir();

        // TODO: 6/15/2016 make this customable
        File annotations = new File(packageRoot, "annotations");
        annotations.mkdir();

        File components = new File(packageRoot, "components");
        components.mkdir();

        File entities = new File(packageRoot, "entities");
        entities.mkdir();

    }
}

File: src/main/java/net/mreunionlabs/tapgradle/generate/generator/JavaStructureGenerator.kt
package net.mreunionlabs.tapgradle.generate.generator
import net.mreunionlabs.tapgradle.generate.GeneratePluginExtension
import net.mreunionlabs.tapgradle.generate.util.PackageDirectoryCreator

import java.io.File

class JavaStructureGenerator {

    //TODO validate that java dir parent has been created
    fun createDirectories(ext: GeneratePluginExtension) {
        val f = File(ext.javaDir)
        f.mkdir()

        val packageString = ext.packageString
        val packageRoot = PackageDirectoryCreator.createPackageDir(f, packageString)

        val pages = File(packageRoot, "pages")
        pages.mkdir()

        // TODO: 6/15/2016 make this customable
        val annotations = File(packageRoot, "annotations")
        annotations.mkdir()

        // Todo create layout component by default
        val components = File(packageRoot, "components")
        components.mkdir()

        // todo create package-info.java in this directory
        val compModel = File(packageRoot, "compModel")
        compModel.mkdir()

        val entities = File(packageRoot, "entities")
        entities.mkdir()

    }
}
